.github/workflows/main.yml

```diff
@@ -0,0 +1,15 @@
+name: CI (main)
+
+on:
+  push:
+    branches: [ "main" ]
+
+jobs:
+  unit-tests:
+    uses: ./.github/workflows/unit-tests.yml
+  publish-caliper:
+    uses: ./.github/workflows/publish.yml
+    needs: unit-tests
+  integration-tests:
```

Comment:
If this is a PR merge then we don't need to run the integration tests. The original azure pipelines just runs unit-tests followed by a publish on a merge